integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/Hooks.java

```diff
@@ -39,42 +38,45 @@ public void before(Scenario scenario) {
-                                bridgeId);
-                        if (processorList.getSize() > 0) {
-                            processorList.getItems().stream().forEach(
-                                    p -> ProcessorResource.deleteProcessor(token, bridgeId,
-                                            p.getId()));
-                            Awaitility.await()
-                                    .atMost(Duration.ofMinutes(2))
-                                    .pollInterval(Duration.ofSeconds(5))
-                                    .until(() -> ProcessorResource.getProcessorList(token, bridgeId)
-                                            .getSize() == 0);
-                        }
-                    }
-                    switch (bridge.getStatus()) {
-                        case ACCEPTED:
-                        case PROVISIONING:
-                        case READY:
-                        case FAILED:
-                            try {
-                                BridgeResource.deleteBridge(token, bridgeId);
-                            } catch (Exception e) {
-                                LOGGER.warn(e, () -> "Unable to delete bridge with id " + bridgeId);
+        if (!Boolean.parseBoolean(DISABLE_CLEANUP)) {
+            String token = Optional.ofNullable(context.getManagerToken()).orElse(BridgeUtils.retrieveBridgeToken());
+            // Remove all bridges/processors created
+            context.getAllBridges().values()
+                    .stream()
+                    .filter(bridgeContext -> !bridgeContext.isDeleted())
+                    .forEach(bridgeContext -> {
+                        final String bridgeId = bridgeContext.getId();
+                        BridgeResponse bridge = BridgeResource.getBridgeDetails(token, bridgeId);
+                        if (bridge.getStatus() == ManagedResourceStatus.READY) {
+                            ProcessorListResponse processorList = ProcessorResource.getProcessorList(
+                                    token,
+                                    bridgeId);
+                            if (processorList.getSize() > 0) {
+                                processorList.getItems().stream().forEach(
+                                        p -> ProcessorResource.deleteProcessor(token, bridgeId,
+                                                p.getId()));
+                                Awaitility.await()
+                                        .atMost(Duration.ofMinutes(2))
+                                        .pollInterval(Duration.ofSeconds(5))
+                                        .until(() -> ProcessorResource.getProcessorList(token, bridgeId)
+                                                .getSize() == 0);
                             }
-                        default:
-                            break;
-                    }
-                });
+                        }
+                        switch (bridge.getStatus()) {
```

Comment:
Is there any state in which the bridge shouldn't be deleted? Is this switch necessary?

Reply:
In case Bridge is in one of deprovisioning statuses (DEPROVISION, DELETING, DELETED) then it is not needed to explicitly delete it.